jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -328,10 +329,13 @@ def _dataset_selected_changed(self, event=None):
             # during initial init, this can trigger before the component is initialized
             return
 
-        selected_spec = self.dataset.selected_spectrum
+        selected_spec = self.dataset.selected_obj
         if selected_spec is None:
             return
 
+        if '_pixel_scale_factor' in selected_spec.meta:
+            self._pixel_scale_factor = selected_spec.meta['_pixel_scale_factor']
```

Comment:
can we just retrieve the value from the metadata when its needed (in `_fit_model_to_spectrum` and `_fit_model_to_cube`) to avoid having to store a copy in the plugin itself?

If this `if` is never entered, then the stored value will be from whatever previous selection did enter this if-statement, which could cause confusing behavior.

Reply:
Good point, updated.